lib/ansible/playbook/role_include.py

```diff
@@ -113,17 +113,17 @@ def load(data, block=None, role=None, task_include=None, variable_manager=None,
             raise AnsibleParserError("'name' is a required field for include_role.")
 
         # validate bad args, otherwise we silently ignore
-        bad_opts = my_arg_names.difference(IncludeRole.VALID_ARGS)
+        bad_opts = my_arg_names.difference(set(IncludeRole.VALID_ARGS))
         if bad_opts:
             raise AnsibleParserError('Invalid options for include_role: %s' % ','.join(list(bad_opts)))
 
         # build options for role includes
-        for key in IncludeRole.FROM_ARGS.intersection(my_arg_names):
+        for key in set(IncludeRole.FROM_ARGS).intersection(my_arg_names):
```

Comment:
same note about not needing to crate another set here with the additional note that you have to reverse the args (since my_arg_names is a set but IncludeRole.FROM_ARGS is not):
``` python
for key in my_arg_names.intersection(IncludeRole.FROM_ARGS):
```